source/extensions/common/wasm/wasm.cc

```diff
@@ -504,6 +504,22 @@ void replaceHeader(Http::HeaderMap* map, absl::string_view key, absl::string_vie
     map->addCopy(lower_key, std::string(value));
 }
 
+const uint8_t* decodeVarint(const uint8_t* start, const uint8_t* end, uint32_t* out) {
+  uint32_t ret = 0;
+  int shift = 0;
+  while (start < end && *start >= 128) {
```

Comment:
Nit: `*start & 0x80`, since you're checking the bit-flag here, as far as I understand.